apps/ecommerce/frontend/src/types.ts

```diff
@@ -98,3 +98,12 @@ export interface DialogInvocationParameters extends KeyValueMap {
   linkType: string;
   fieldType: FieldType;
 }
+
+// the variant states of ProductCard, for now, labeled by location
+export type ProductCardType = 'field' | 'dialog';
```

Comment:
this for now is a simpler way to designate card variants, this will most likely me adjusted as we continue to figure out the product card mapping on the frontend.